src/app/search/search.component.ts

```diff
@@ -161,16 +162,25 @@ export class SearchComponent implements OnInit, OnDestroy {
     this.filterKeys$ = this.searchQuery.filterKeys$;
     this.suggestTerm$ = this.searchQuery.suggestTerm$;
     this.selectedIndex$ = this.searchQuery.savedTabIndex$;
-    // Initialize mappings
-    this.bucketStubs = this.searchService.initializeCommonBucketStubs();
-    this.friendlyNames = this.searchService.initializeFriendlyNames();
-    this.entryOrder = this.searchService.initializeEntryOrder();
-    this.toolTips = this.searchService.initializeToolTips();
-    this.expandedPanels = this.searchService.initializeExpandedPanels();
-
+    this.initializeMappings(0);
     this.clearFacetSearches();
   }
 
+  private tabIndex: number | null = null;
+
+  initializeMappings(newTabIndex: number) {
+    if (newTabIndex !== this.tabIndex) {
+      this.tabIndex = newTabIndex;
+      console.log('generate new mappings');
```

Comment:
Debug message

Reply:
ack, I thought I took care of all those :)